### ISSUE TITLE
type guards with generic types in switch expression 

### ISSUE THREAD

**Issue by smazzoleni** [eclipse/xtext-xtend#268]
Hi. I'm very new to Xtend and while writing learning tests, i found following assertion failed
```
assertThat(
	switch new ArrayList<String> {
		List<Integer>: "list of integers"
		List<String>: "list of strings"
		default: "something else"
	}
).isEqualTo("list of strings")
```
expected :<"list of [string]s"> but was:<"list of [integer]s">

**Comment by cdietrich** [comment]
typeguards wont work with generics https://bugs.eclipse.org/bugs/show_bug.cgi?id=342825

**Comment by cdietrich** [comment]
=> we need a warning that flags the code as stated in linked bugzilla

**Comment by cdietrich** [comment]
Fixed in Xtext 2.13
